src/libcore/slice/mod.rs

```diff
@@ -400,14 +400,14 @@ impl<T> SliceExt for [T] {
         while size > 1 {
             let half = size / 2;
             let mid = base + half;
-            // mid is always in [0, size).
+            // mid is always in [0, size), that means mid is >= 0 and < size.
             // mid >= 0: by definition
             // mid < size: mid = size / 2 + size / 4 + size / 8 ...
             let cmp = f(unsafe { s.get_unchecked(mid) });
             base = if cmp == Greater { base } else { mid };
             size -= half;
         }
-        // base is always in [0, size) because base <= mid.
+        // base is always in [0, size) because base <= mid and base < size.
```

Comment:
Hm, as written that isn't exactly clarifying. Let's just drop it &mdash; you're the one who was confused by this and evidently you didn't see the need to clarify this line.

Reply:
Drop the PR or this line?

Reply:
This line. The other change is great :)